src/strands_tools/bright_data.py

```diff
@@ -0,0 +1,598 @@
+"""
+Tool for web scraping, searching, and data extraction using Bright Data for Strands Agents
+
+This module provides comprehensive web scraping and data extraction capabilities using
+Bright Data as the backend. It handles all aspects of web scraping with a user-friendly
+interface and proper error handling.
+
+Key Features:
+------------
+1. Web Scraping:
+   • scrape_as_markdown: Scrape webpage content and return as Markdown
+   • get_screenshot: Take screenshots of webpages
+   • search_engine: Perform search queries using various search engines
+   • web_data_feed: Extract structured data from websites like LinkedIn, Amazon, Instagram, etc.
+
+2. Advanced Capabilities:
+   • Support for multiple search engines (Google, Bing, Yandex)
+   • Advanced search parameters including language, location, device type
+   • Extracting structured data from various websites
+   • Screenshot generation for web pages
+
+3. Error Handling:
+   • Graceful API error handling
+   • Clear error messages
+   • Timeout management for web_data_feed
+
+Usage Examples:
+--------------
+```python
+from strands import Agent
+from strands_tools import bright_data
+
+agent = Agent(tools=[bright_data])
+
+# Scrape webpage as markdown
+agent.tool.bright_data(
+    action="scrape_as_markdown",
+    url="https://example.com"
+)
+
+# Search using Google
+agent.tool.bright_data(
+    action="search_engine",
+    query="climate change solutions",
+    engine="google",
+    country_code="us",
+    language="en"
+)
+
+# Extract product data from Amazon
+agent.tool.bright_data(
+    action="web_data_feed",
+    source_type="amazon_product",
+    url="https://www.amazon.com/product-url"
+)
+```
+"""
+
+import json
+import logging
+import os
+import time
+from typing import Any, Dict, Optional
+from urllib.parse import quote
+
+import requests
+from rich.console import Console
+from rich.panel import Panel
+from rich.text import Text
+from strands.types.tools import ToolResult, ToolResultContent, ToolUse
+
+logger = logging.getLogger(__name__)
+
+console = Console()
```

Comment:
Can we use "from strands_tools.utils import console_util" for consistency across tools?